plugin-server/src/utils/db/db.ts

```diff
@@ -878,19 +878,29 @@ export class DB {
         return client ? kafkaMessages : updatedPerson
     }
 
-    public async deletePerson(person: Person, client: PoolClient): Promise<ProducerRecord[]> {
-        await client.query('DELETE FROM posthog_person WHERE team_id = $1 AND id = $2', [person.team_id, person.id])
-        const kafkaMessages = [
-            generateKafkaPersonUpdateMessage(
-                person.created_at,
-                person.properties,
-                person.team_id,
-                person.is_identified,
-                person.uuid,
-                null,
-                1
-            ),
-        ]
+    public async deletePerson(person: Person, client?: PoolClient): Promise<ProducerRecord[]> {
+        const result = await this.postgresQuery<{ version: string }>(
+            'DELETE FROM posthog_person WHERE team_id = $1 AND id = $2 RETURNING version',
+            [person.team_id, person.id],
+            'deletePerson',
+            client
+        )
+
+        let kafkaMessages: ProducerRecord[] = []
+
+        if (result.rows.length > 0) {
+            kafkaMessages = [
+                generateKafkaPersonUpdateMessage(
+                    person.created_at,
+                    person.properties,
+                    person.team_id,
+                    person.is_identified,
+                    person.uuid,
+                    Number(result.rows[0].version || 0) + 1,
+                    1
```

Comment:
An update could overwrite this (if one happens to run in parallel, after this postgres call), in the update path we should either: (1) not change the `is_deleted` value, (2) not run the update if is_deleted is set (i.e. use `upsert` there)

Reply:
That's a separate bug and not in scope to be fixed in this PR. Let's keep this scoped down tightly.

Mind opening a bug ticket?

Reply:
Lets just put up a fix pr, I can do that (agree it's separate, I just only thought about it when reviewing your PR)

Reply:
Looked into this & thought about this - if the person is deleted they aren't in Postgres anymore, so update would fail in `updatePersonDeprecated` function, which would mean we error and don't end up sending the update to CH. The `version +100` solves the problem, where the postgres update happened first, then postgres delete happened and we are writing both messages to CH. So this is handled.